rosbag2_transport/src/rosbag2_transport/rosbag2_transport_python.cpp

```diff
@@ -69,7 +71,7 @@ rosbag2_transport_record(PyObject * Py_UNUSED(self), PyObject * args, PyObject *
   bool include_hidden_topics = false;
   if (
     !PyArg_ParseTupleAndKeywords(
-      args, kwargs, "ssssss|bbKKKOb", const_cast<char **>(kwlist),
+      args, kwargs, "ssssss|bbKKKObs", const_cast<char **>(kwlist),
```

Comment:
We will need to be passing these QoS profiles around as actual objects. Are you familiar with the "capsule" concept used in Python<->C bindings to hold opaque C pointers?

If not yet, let's discuss. I don't know off the top of my head what the implementation looks like, but we should be able to pass either:
* a map of topic -> rmw_qos_profile_t
* or a list/array of pairs of (topic, rmw_qos_profile_t)

That way we've already deserialized here and the rosbag2_transport side can just receive structured data.

Reply:
Familiar with Python's C-API [Object protocol](https://docs.python.org/3/c-api/object.html).

I was thinking something along the lines of:

```
Rosbag2Qos qos{};
qos.history = *PyString_AsString(PyMapping_GetItemString(pDict, "history"))
...
```

Reply:
Also though on using [`pybind`](https://pybind11.readthedocs.io/en/master/advanced/cast/stl.html) now that it's part of the repo?